sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BooleanLiteralUnnecessary.cs

```diff
@@ -104,4 +107,16 @@ private void M()
             }
         }
     }
+
+    public class SocketContainer
+    {
+        private IEnumerable<Socket> sockets;
+        public bool IsValid =>
+            sockets.All(x => x.IsStateValid == true); // Compliant, this failed when we compile with Roslyn 1.x
```

Comment:
The problem is not reproducible in the unit tests because we compile with Roslyn 2+ and I suspect this was caused by a bug in Roslyn itself. The only way I found to test is to reference the SonarAnalyzer.CSharp nuget in a VS2015 project and paste the SocketContainer/Socket classes.